### PR TITLE
[Doc] Fix refresh external table doc

### LINKED ISSUE #14155: [HIVE-METADATA] refresh external table [partition(***=***)] error after insert overwrite partition
### Steps to reproduce the behavior (Required)
```
5 rows in hive table
(3, Decimal('325.15'), datetime.datetime(2021, 12, 1, 13, 25, 36), 'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?', True, '2001-01-01')
(4, Decimal('365.54'), datetime.datetime(2001, 3, 25, 3, 44, 43), 'Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.', False, '2002-01-01')
(2, Decimal('365.54'), datetime.datetime(2001, 3, 25, 3, 44, 43), 'Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.', False, '2007-06-07')
(5, Decimal('365.54'), datetime.datetime(2019, 8, 30, 23, 25, 43), 'old partition', False, '2007-06-07')
(1, Decimal('325.15'), datetime.datetime(2021, 12, 1, 13, 25, 36), 'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?', True, '2025-10-15')
select * from hive_metadata_d1033e3a_6d88_11ed_a9c2_00163e0e550b.extbl_hive_hdfs_orc_par order by col_date,col_tinyint;
4
refresh external table hive_metadata_d1033e3a_6d88_11ed_a9c2_00163e0e550b.extbl_hive_hdfs_orc_par partition('2007-06-07');
select * from hive_metadata_d1033e3a_6d88_11ed_a9c2_00163e0e550b.extbl_hive_hdfs_orc_par order by col_date,col_tinyint;
5
refresh external table hive_catalog_d103b338_6d88_11ed_a9c2_00163e0e550b.hive_metadata_test_d103b23e_6d88_11ed_a9c2_00163e0e550b.hive_hdfs_orc_par partition('2007-06-07');
select * from hive_catalog_d103b338_6d88_11ed_a9c2_00163e0e550b.hive_metadata_test_d103b23e_6d88_11ed_a9c2_00163e0e550b.hive_hdfs_orc_par order by col_date,col_tinyint;
5
insert overwrite table hive_hdfs_orc_par partition(col_date = '2007-06-07') select 99,654854.54,'2007-12-01 23:24:25','insert overwrite partitioin',true
4 rows in hive table
(3, Decimal('325.15'), datetime.datetime(2021, 12, 1, 13, 25, 36), 'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?', True, '2001-01-01')
(4, Decimal('365.54'), datetime.datetime(2001, 3, 25, 3, 44, 43), 'Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.', False, '2002-01-01')
(99, Decimal('654854.54'), datetime.datetime(2007, 12, 1, 23, 24, 25), 'insert overwrite partitioin', True, '2007-06-07')
(1, Decimal('325.15'), datetime.datetime(2021, 12, 1, 13, 25, 36), 'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?', True, '2025-10-15')
select * from hive_metadata_d1033e3a_6d88_11ed_a9c2_00163e0e550b.extbl_hive_hdfs_orc_par order by col_date,col_tinyint;
execute sql error and messages: (1064, 'HdfsOrcScanner::do_open failed. reason = Failed to read hdfs://emr-header-1.cluster-49091:9000/user/hive/warehouse/hive_metadata_test_d103b23e_6d88_11ed_a9c2_00163e0e550b.db/hive_hdfs_orc_par/col_date=2007-06-07/000000_0: IO error: cannot read fully\n/root/starrocks/be/src/exec/vectorized/hdfs_scanner.cpp:41 _stream->read_at_fully(offset, data, size):file = hdfs://emr-header-1.cluster-49091:9000/user/hive/warehouse/hive_metadata_test_d103b23e_6d88_11ed_a9c2_00163e0e550b.db/hive_hdfs_orc_par/col_date=2007-06-07/000000_0'), and time is: 2022-11-26 20:52:17
refresh external table hive_metadata_d1033e3a_6d88_11ed_a9c2_00163e0e550b.extbl_hive_hdfs_orc_par partition('2007-06-07','2001-01-01');
execute sql error and messages: (1064, 'Unexpected exception: null value in entry: 2001-01-01=null'), and time is: 2022-11-26 20:52:17
```

### Expected behavior (Required)
refresh table success

### Real behavior (Required)
refresh table error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.5-RELEASE 8128202 |
+----------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14155

## Problem Summary(Required) ：

**refresh external table just support:**
    REFRESH EXTERNAL TABLE table_name 
    [PARTITION ('partition_name=partition_value', ...)];
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-definition/REFRESH EXTERNAL TABLE.md [hunk 1]
```diff
@@ -22,19 +22,20 @@ The following describes the syntaxes and parameters based on different cases:
 
     ```SQL
     REFRESH EXTERNAL TABLE table_name 
-    [PARTITION ('partition_name', ...)];
+    [PARTITION ('partition_name=partition_value', ...)];
```
Comment (stephen-shelby) [inline docs/sql-reference/sql-statements/data-definition/REFRESH EXTERNAL TABLE.md:25]: remove it

**Review by stephen-shelby — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by evelynzhaojie** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
@evelynzhaojie admit_test done,  pls merge it thanks

**Comment by evelynzhaojie** [comment]
@mergify backport branch-2.5 branch-2.4

**Pull request merged**